### ISSUE TITLE
bug: `delete` sanity check failed in materialize

### ISSUE THREAD

**Issue by xxchan** [risingwavelabs/risingwave#5913]
### Describe the bug

(e2e parallel)
https://buildkite.com/risingwavelabs/pull-request/builds/10459#0183eca8-be29-49c7-913b-b4ec8bbfd674

```
thread 'risingwave-streaming-actor' panicked at 'inconsistent delete!
table_id: 2128, vnode: 38, key: Row([Some(Int64(0)), Some(Int64(12))])
stored value: None
expected value: Row([Some(Int64(12)), Some(Int64(0))])', /risingwave/src/storage/src/table/streaming_table/state_table.rs:676:13
stack backtrace:
2022-10-18T20:06:48.764903Z DEBUG risingwave_stream::task::stream_manager: drop actors actors=[2442, 2443, 2444, 2445, 2454, 2455, 2456, 2457]
   0: rust_begin_unwind
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/core/src/panicking.rs:142:14
   2: risingwave_storage::table::streaming_table::state_table::StateTable<S>::do_delete_sanity_check::{{closure}}
             at ./src/storage/src/table/streaming_table/state_table.rs:676:13
   3: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/core/src/future/mod.rs:91:19
   4: risingwave_storage::table::streaming_table::state_table::StateTable<S>::batch_write_rows::{{closure}}
             at ./src/storage/src/table/streaming_table/state_table.rs:614:70
   5: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/core/src/future/mod.rs:91:19
   6: risingwave_storage::table::streaming_table::state_table::StateTable<S>::commit::{{closure}}
             at ./src/storage/src/table/streaming_table/state_table.rs:570:57
   7: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/9067d5277d10f0f32a49ec9c125a33828e26a32b/library/core/src/future/mod.rs:91:19
   8: risingwave_stream::executor::mview::materialize::MaterializeExecutor<S>::execute_inner::{{closure}}
             at ./src/stream/src/executor/mview/materialize.rs:135:54
```

**Comment by BugenZhao** [comment]
All `Materialize`s have an upstream of hash agg, so I guess it's related to #5911. cc @chenzl25

**Comment by chenzl25** [comment]
> All `Materialize`s have an upstream of hash agg, so I guess it's related to #5911. cc @chenzl25

Yes, I find a bug introduced #5882 and try to fix it by #5914.
